skyvern/webeye/real_browser_state.py

```
@@ -0,0 +1,435 @@
+from skyvern.webeye.utils.page import ScreenshotMode, SkyvernFrame
+
+LOG = structlog.get_logger()
+
+
+class RealBrowserState(BrowserState):
+    def __init__(
+        self,
+        pw: Playwright,
+        browser_context: BrowserContext | None = None,
+        page: Page | None = None,
+        browser_artifacts: BrowserArtifacts = BrowserArtifacts(),
```

Ruff (0.14.7)

Do not perform function call `BrowserArtifacts` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)